### ISSUE TITLE
Licence for CMSSW still unclear

### ISSUE THREAD

**Issue by RaoOfPhysics** [cernopendata/opendata.cern.ch#1232]
- The two CMSSW records on CODP mention the licence as being `Apache License, Version 2.0`:
    - http://opendata.cern.ch/record/221
    - http://opendata.cern.ch/record/220
- However, the CMSSW repo on GitHub does not have a valid `LICENSE` file for the whole project: https://github.com/cms-sw/cmssw
    - Although, some sub-directories and files seem to be accompanied by licences: https://github.com/cms-sw/cmssw/search?utf8=%E2%9C%93&q=license (Some Apache, some GPL, etc.)
    - **Question**: Can the whole of CMSSW be licensed under Apache when some components are GPL?
- When contacted in 2014-09, @davidlange6 said,
> “The CB has endorsed Apache License, Version 2.0 - but strictly following the requirements of that is not yet achieved [for CMSSW].”
- When I mentioned this in #71, it may have been interpreted as CMSSW having a confirmed licence, which isn't the case.
- **How should we proceed?** At the moment, those directories/files without licences technically have all rights reserved -- you, you can view the source (“open source“) but they are not free to use/modify.
    - We need confirmation of the licence of CMSSW as a whole.

cc: @tiborsimko, @katilp, @ekenn003

**Comment by davidlange6** [comment]
> On Mar 10, 2017, at 4:24 PM, RaoOfPhysics <notifications@github.com> wrote:
> 
> 	• The two CMSSW records on CODP mention the licence as being Apache License, Version 2.0:
> 		• http://opendata.cern.ch/record/221
> 		• http://opendata.cern.ch/record/220
> 	• However, the CMSSW repo on GitHub does not have a valid LICENSE file for the whole project: https://github.com/cms-sw/cmssw
> 		• Although, some sub-directories and files seem to be accompanied by licences: https://github.com/cms-sw/cmssw/search?utf8=%E2%9C%93&q=license (Some Apache, some GPL, etc.)
> 		• Question: Can the whole of CMSSW be licensed under Apache when some components are GPL?
> 	• When contacted in 2014-09, @davidlange6 said,
> “The CB has endorsed Apache License, Version 2.0 - but strictly following the requirements of that is not yet achieved [for CMSSW].”
> 

and indeed, the context of that discussion was about analysis code examples and less about CMSSW as a whole.

As I mentioned at the time, in addition to the code license questions (which are still not solved), there are additional issues if you really mean to license a VM with libraries of  CMSSW+external dependences.

**Comment by katilp** [comment]
@davidlange6 : @RaoOfPhysics refers here to a proposal that was discussed and approved at the CB meeting, independently from the Open Data, and to my understanding it is intended to CMSSW as a whole

https://indico.cern.ch/event/143964/ (CMS internal)
https://indico.cern.ch/event/143964/contributions/1378788/attachments/133233/189074/LicensingCB77.pdf (CMS internal)
(In the minutes: "8. The CB approved the proposal for Copyright and licensing of CMSSW software" )

**Comment by davidlange6** [comment]
Right-  thats for the code (and while proposed and discussed, the implementation of the specifics has unfortunately not been completed as I said)

**Comment by davidlange6** [comment]
In any case, "CMSSW as a whole" means different things to different people. Its good to be more specific.

**Comment by RaoOfPhysics** [comment]
@davidlange6: Thanks for joining the discussion! :)

> In any case, "CMSSW as a whole" means different things to different people. Its good to be more specific.

Indeed, this issue was triggered by the fact that as part of the open-data efforts, we have also been pointing to the relevant versions of CMSSW from the CERN Open Data Portal (records `220` and `221`, as indicated in the OP). There, we erroneously state that the licence is Apache. Should we therefore simply remove reference of a licence or do we say "components licensed modularly"?

I suppose part of the concern is that while we can point to CMSSW we cannot in good faith ask users to use the software (let alone re-use/modify it) in the absence of some formal statement permitting them to do so (through some licence). Perhaps @tiborsimko or @TimSmithCH can chime in?

> […] there are additional issues if you really mean to license a VM with libraries of CMSSW+external dependences.

This will also need to be addressed if we are to adhere to the standards/practices of open research.

**Comment by davidlange6** [comment]
> On Mar 13, 2017, at 9:22 AM, RaoOfPhysics <notifications@github.com> wrote:
> 
> @davidlange6: Thanks for joining the discussion! :)
> 
> In any case, "CMSSW as a whole" means different things to different people. Its good to be more specific.
> 
> Indeed, this issue was triggered by the fact that as part of the open-data efforts, we have also been pointing to the relevant versions of CMSSW from the CERN Open Data Portal (records 220 and 221, as indicated in the OP). There, we erroneously state that the licence is Apache. Should we therefore simply remove reference of a licence or do we say "components licensed modularly"?

again, what is CMSSW and a "version of CMSSW" precisely?

**Comment by katilp** [comment]
@davidlange6 In my simple-minded view: for the versions we recommend with the open data,
CMSSW is the code that can be found in the repositories:
https://github.com/cms-sw/cmssw/tree/CMSSW_4_2_X
https://github.com/cms-sw/cmssw/tree/CMSSW_5_3_X

**Comment by davidlange6** [comment]
so you mean code in the CMSSW repository. There (afaik) we intend (and desire) to have an apache-like license but have not gotten there yet.

**Comment by katilp** [comment]
The CMS collaboration board approved at the CB meeting on February 8, 2019: 

> Adoption	 of	 Apache-2	 license for	 software	 developed	 in	 CMS,	 unless	 other	 licenses	apply
